src/main/java/org/fogbowcloud/manager/core/plugins/compute/openstack/OpenStackNovaV2ComputePlugin.java

```diff
@@ -0,0 +1,438 @@
+
+    private static final Logger LOGGER = Logger.getLogger(OpenStackNovaV2ComputePlugin.class);
+
+    private List<Flavor> flavors;
+    private Properties properties;
+    private HttpClient client;
+
+    public OpenStackNovaV2ComputePlugin(Properties properties) {
+        this.flavors = new ArrayList<>();
+        this.properties = properties;
+        this.initClient();
+    }
+
+    public String requestInstance(ComputeOrder computeOrder, String imageId) throws RequestException {
+        LOGGER.debug("Requesting instance with token=" + computeOrder.getLocalToken());
+
+        Token localToken = computeOrder.getLocalToken();
+        String flavorId = updateAndFindSmallestFlavor(computeOrder).getId();
+        String tenantId = getTenantId(localToken);
+        String networkId = getNetworkId();
+        String userData = computeOrder.getUserData().getContent();
+        String keyName = getKeyName(tenantId, localToken, computeOrder.getPublicKey());
+        String endpoint = getComputeEndpoint(tenantId, SERVERS);
+
+        try {
+            JSONObject json = generateJsonRequest(imageId, flavorId, userData, keyName, networkId);
+            String jsonResponse = doPostRequest(endpoint, localToken, json);
+
+            return getAttFromJson(ID_JSON_FIELD, jsonResponse);
+        } catch (JSONException e) {
+            LOGGER.error(e);
+            throw new RequestException(ErrorType.BAD_REQUEST, ResponseConstants.IRREGULAR_SYNTAX);
+        } finally {
+            if (keyName != null) {
+                try {
+                    deleteKeyName(tenantId, localToken, keyName);
+                } catch (Throwable t) {
+                    LOGGER.warn("Could not delete key.", t);
+                }
+            }
+        }
+    }
+
+    private void initClient() {
+        HttpRequestUtil.init(properties);
+        this.client = HttpRequestUtil.createHttpClient();
+    }
+
+    protected String getTenantId(Token localToken) {
+        return localToken.getAttributes().get(TENANT_ID);
+    }
+
+    protected String getNetworkId() {
+        return properties.getProperty(COMPUTE_NOVAV2_NETWORK_KEY);
+    }
+
+    protected String getKeyName(String tenantId, Token localToken, String publicKey) throws RequestException {
+        String keyname = null;
+
+        if (publicKey != null && !publicKey.isEmpty()) {
+            String osKeypairEndpoint = getComputeEndpoint(tenantId, SUFFIX_ENDPOINT_KEYPAIRS);
+
+            keyname = UUID.randomUUID().toString();
+            JSONObject keypair = new JSONObject();
+
+            try {
+                keypair.put(NAME_JSON_FIELD, keyname);
+                keypair.put(PUBLIC_KEY_JSON_FIELD, publicKey);
+                JSONObject root = new JSONObject();
+                root.put(KEYPAIR_JSON_FIELD, keypair);
+                doPostRequest(osKeypairEndpoint, localToken, root);
+            } catch (JSONException e) {
+                LOGGER.error(e);
```

Comment:
Please write a LOGGER message, it could help to debug any possible error at this method.

Reply:
Fixed!